File: services/ai-service/app/kafka/topics.py
"""
@file topics.py
@description
Kafka topic name constants for the AI Recommendation Service.

These must stay in sync with the Node.js core-api Kafka topics
(see services/core-api/src/kafka/topics.js).
"""

RAW_SUPPLIER_FEEDS: str = "raw_supplier_feeds"
DEALS_NORMALIZED: str = "deals.normalized"
DEALS_SCORED: str = "deals.scored"
DEALS_TAGGED: str = "deals.tagged"
DEAL_EVENTS: str = "deal.events"

BOOKING_EVENTS: str = "booking.events"
USER_ACTIVITY: str = "user.activity"

ALL_TOPICS = (
    RAW_SUPPLIER_FEEDS,
    DEALS_NORMALIZED,
    DEALS_SCORED,
    DEALS_TAGGED,
    DEAL_EVENTS,
    BOOKING_EVENTS,
    USER_ACTIVITY,
)

File: services/core-api/src/kafka/topics.js
/**
 * @file topics.js
 * @description
 * Central definition of Kafka topic names used by the platform.
 *
 * Keeping topic names in one place:
 * - Avoids typos across producers/consumers.
 * - Makes it easy to document and evolve the messaging topology.
 */

export const RAW_SUPPLIER_FEEDS = "raw_supplier_feeds";
export const DEALS_NORMALIZED = "deals.normalized";
export const DEALS_SCORED = "deals.scored";
export const DEALS_TAGGED = "deals.tagged";
export const DEAL_EVENTS = "deal.events";

export const BOOKING_EVENTS = "booking.events";
export const USER_ACTIVITY = "user.activity";

/**
 * Frozen map of all topic names for convenience.
 */
export const KAFKA_TOPICS = Object.freeze({
  RAW_SUPPLIER_FEEDS,
  DEALS_NORMALIZED,
  DEALS_SCORED,
  DEALS_TAGGED,
  DEAL_EVENTS,
  BOOKING_EVENTS,
  USER_ACTIVITY,
});

export default KAFKA_TOPICS;
